qemu/tests/apicv_test.py

```diff
@@ -0,0 +1,108 @@
+import re
+import os
+import time
+import logging
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import env_process
+from virttest import data_dir
+from virttest import utils_net
+from virttest import utils_netperf
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    apicv test:
+    1) Check if apicv is enabled on host, if not, enable it
+    2) Boot guest and run netperf inside guest
+    3) Record throughput and shutdown guest
+    4) Disable apicv on host
+    5) Boot guest and run netperf inside guest again
+    6) Compare benchmark scores with step 3)
+    7) Restore env, set apicv back
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def reload_module(value):
+        """
+        Reload module
+        """
+        process.system("rmmod %s" % module)
+        cmd = "modprobe %s %s=%s" % (module, mod_param, value)
+        process.system(cmd)
+
+    def run_netperf():
+        """
+        Run netperf test, return average throughput
+        """
+        error_context.context("Run netperf test", logging.info)
+        n_server.start()
+        n_client.session = session
```

Comment:
Will it be rebooted? If no, I think this isn't necessary.

Reply:
Yes, guest recreate after disable apicv.